### PR TITLE
fix(common): clean up types

### PR DESCRIPTION
**Pull request by eunjae-lee**
## What?

This PR does several things:

- move `AppConfig` patch to `index.d.ts`, so that all the projects that use nuxt-base will have the same type.
- configure `./types` to be auto-imported
- remove unnecessary type imports

### CODE REVIEW

File: space-plugins/nuxt-starter/app.config.ts [hunk 1]
```diff
@@ -1,14 +1,8 @@
 export default defineAppConfig({
+	type: 'space-plugin',
```
Comment (eunjae-lee) [inline space-plugins/nuxt-starter/app.config.ts:2]: This was outdated. We've moved `type` to the top level of app config in a recent PR.

File: space-plugins/nuxt-starter/app.config.ts [hunk 2]
```diff
@@ -1,14 +1,8 @@
...
 		oauth: true,
 		origin: 'https://app.storyblok.com',
 	},
-	auth: {
-		endpointPrefix: '/api/connect',
-		initOauthFlowUrl: `/api/connect/storyblok`,
-		successCallback: '/',
-		errorCallback: '/401',
-	},
 });
```
Comment (eunjae-lee) [inline space-plugins/nuxt-starter/app.config.ts:8]: Now we no longer need to have full appConfig. Only things that we want to override.

File: space-plugins/nuxt-base/index.d.ts [hunk 3]
```diff
@@ -6,4 +6,18 @@ declare module 'h3' {
...
+		appBridge?: AppBridgeConfig;
+		auth?: AuthConfig;
+	}
+
+	interface AppConfig {
+		type: PluginType;
+		appBridge: AppBridgeConfig;
+		auth: AuthConfig;
+	}
+}
```
[the remark this answers is not in the document]
Comment (BibiSebi) [reply space-plugins/nuxt-base/index.d.ts:21]: that makes sense maybe adding a comment would help us understand this better in future? WDYT?

**Review by BibiSebi — APPROVED** [review]
Looking great 💪 thank you for the speedy adjustments 🙏  

Left one small comment.

**Pull request merged**